font/src/ft/mod.rs

```diff
@@ -253,31 +253,23 @@ impl FreeTypeRasterizer {
         let primary_font = pattern.render_prepare(config, primary_font);
 
         // Hash pattern together with request pattern to include requested font size in the hash
-        let primary_font_id = FontID::new(hash, primary_font.hash());
+        let primary_font_key = FontKey::from_pattern_hashes(hash, primary_font.hash());
 
         // Reload already loaded faces and drop their fallback faces
-        let font_key = if let Some(font_key) = self.keys.remove(&primary_font_id) {
-            let fallback_list = self.fallback_lists.remove(&font_key).unwrap_or_default();
+        if self.faces.remove(&primary_font_key).is_some() {
+            let fallback_list = self.fallback_lists.remove(&primary_font_key).unwrap_or_default();
 
             for fallback_font in &fallback_list.list {
-                if let Some(ff_key) = self.keys.get(&fallback_font.id) {
-                    // Skip primary fonts, since these are all reloaded later
-                    if !self.fallback_lists.contains_key(&ff_key) {
-                        self.faces.remove(ff_key);
-                        self.keys.remove(&fallback_font.id);
-                    }
+                // Skip primary fonts, since these are all reloaded later
+                if !self.fallback_lists.contains_key(&fallback_font.key) {
+                    self.faces.remove(&fallback_font.key);
                 }
             }
-
-            let _ = self.faces.remove(&font_key);
-            Some(font_key)
-        } else {
-            None
-        };
+        }
```

Comment:
The `primary_font_key` is a hash that uniquely identifies a font, since it's just a combinations of all patterns required to load it.

So it seems like we should just be able to return the `FontKey` immediately when we find that it is already loaded? Or what do we reload the fonts for?